addons/account/migrations/9.0.1.1/pre-migration.py

```diff
@@ -165,8 +166,37 @@ def map_account_tax_template_type(cr):
         table='account_tax_template', write='sql')
 
 
-@openupgrade.migrate()
-def migrate(cr, version):
+def blacklist_field_recomputation(env):
+    """Create computed fields that take long time to compute, but will be
+    filled with valid values by migration."""
+    from openerp.addons.account.models.account_move import \
+            AccountMove, AccountMoveLine
+    AccountMove._openupgrade_recompute_fields_blacklist = [
+        'currency_id',
+        'amount',
+        'matched_percentage',
+    ]
+    AccountMoveLine._openupgrade_recompute_fields_blacklist = [
+        'amount_residual',
```

Comment:
amount_residual since some time is already filled in the post_migration of account_full_reconcile. So I do not think this can be the problem.